Mathlib/CategoryTheory/Category/Cat/Relation.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2024 Nicolas Rolland. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nicolas Rolland
+-/
+import Mathlib.CategoryTheory.Category.Cat
+import Mathlib.CategoryTheory.ConnectedComponents
+
+/-!
+# Relation induced by a category
+
+The hom-set of a category can be seen as a (proof relevant) relation on its objects :
+Two objects are connected if there is an arrow between them.
+This relation is not an equivalence but can be turned into one.
+
+## Equivalence relation induced by a category
+
+One can take the equivalence closure, under which two objects are connected
+iif there is a zigzag of arrows between them.
+
+As a relation, it is proof irrelavant, in the sense that it does not know by which specific zigzag
+two elements are connected, only that they are.
+
+## Implmentation notes
+
+We rely on `CategoryTheory.ConnectedComponents` and not on `Quiver.ConnectedComponent`
+
+-/
+
+namespace CategoryTheory.Cat
+
+variable {C D : Cat}
+variable {a b : C}
+variable (F : C ⥤ D)
+
+open Relation
+
+private abbrev zigzagSetoidC : Setoid C := Zigzag.setoid C
```

Comment:
This can be replaced by `Zigzag.setoid C` everywhere it appears in the code, except in the definition of `ccfmap` where it causes an error in the line below. I'm not sure why.